tests/php/Control/SessionTest.php

```diff
@@ -335,4 +335,35 @@ public function testSave()
             $_SESSION
         );
     }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testSessionOutput()
```

Comment:
Isn't an additional issue here that the first request that requires session should set a cookie, and then subsequent requests should *not* re-set the same cookie unless the info in the cookie has changed?

This test only seems to check that a single cookie is sent one time.

Reply:
That's correct, i can only test the first scenario, as session_start emits the cookie, You can't actively test session resume in phpunit, as you'd need to start a php session, and then resume it in a new  thread?  - To my knowledge i don't think that's possible.

Reply:
You can do functional tests though still, right? Extend `FunctionalTest`, then make multiple calls to `$this->get('/')` and inspect the responses each time to see what the header data is...

Having said that, I'm not sure if that will work as it doesn't do a full request/response send I don't think (so PHP won't necessarily force-send its own cookie headers via `session_start()`...

Reply:
This feels like the server behaviour testing rather then unit. I'm not sure if it's worth it, but we potentially could consider testing it in behat tests.

Reply:
Yes, there's little point in unit testing this, as we can only test one of the behaviours. We can't test session resume functionality as it's core PHP functionality. We can't test this from behat either, as the session flag `httponly` is set, so session cookies aren't readable from javascript (for security reasons).  I wouldn't recommend overloading that just for testings sake. 

The only way to really test the resume behaviour is make a request using curl or similar.

```
 // should set session cookie
curl -svo /dev/null <my-url>

// should reply without a cookie parameter as the session is resumed.
curl -svo /dev/null <my-url> --cookie "mycookiename=<mycookievalue>;"

```

Any other ideas? or do we just accept it as is? @dnsl48 @madmatt

Reply:
I didn't dig into it yet, but I think we use `chromedriver` for behat, which may give us access to httponly cookies.

Reply:
Ok, I've investigated it and it appears we don't actually use ChromeDriver, but rather our own implementation named `FacebookWebDriver` and facebook `RemoteWebDriver`, which seem to be custom implementations of `Selenium2` driver.

Here's the table with capabilities of drivers: http://mink.behat.org/en/latest/guides/drivers.html

Unless we migrate to ChromeDriver we won't be able to access headers. However, I think that's obviously not within the context of this issue.